symengine/tests/basic/test_sets.cpp

```diff
@@ -61,6 +69,43 @@ TEST_CASE("Interval : Basic", "[basic]")
     REQUIRE(eq(*r1->contains(one), *boolTrue));
     r4 = interval(zero, i2); // [0, 2]
     REQUIRE(eq(*r3, *r4));
+
+    r3 = r1->set_complement(reals);
+    r4 = interval(NegInf, zero, true, true)
+             ->set_union(interval(i20, Inf, true, true));
+    REQUIRE(eq(*r3, *r4));
+    r3 = (interval(im5, i2, true, false))->set_complement(reals);
+    r4 = interval(NegInf, im5, true, false)
+             ->set_union(interval(i2, Inf, true, true));
+    REQUIRE(eq(*r3, *r4));
+    r3 = (interval(im5, i2, false, true))->set_complement(reals);
+    r4 = interval(NegInf, im5, true, true)
+             ->set_union(interval(i2, Inf, false, true));
+    REQUIRE(eq(*r3, *r4));
+    r3 = (interval(im5, i2, true, true))->set_complement(reals);
+    r4 = interval(NegInf, im5, true, false)
+             ->set_union(interval(i2, Inf, false, true));
+    REQUIRE(eq(*r3, *r4));
+
+    r3 = (interval(im5, i2))->set_complement(interval(im5, i2, true, true));
+    REQUIRE(eq(*r3, *emptyset()));
+
+    r3 = (interval(im5, i2, true, true))->set_complement(interval(im5, i2));
+    r4 = finiteset({im5, i2});
+    REQUIRE(eq(*r3, *r4));
+
+    r3 = (interval(im5, i2, true, false))->set_complement(interval(im5, i20));
+    r4 = finiteset({im5})->set_union(interval(i2, i20, true, false));
+    REQUIRE(eq(*r3, *r4));
+
+    r3 = finiteset({symbol("x"), symbol("y"), symbol("z")});
+    r4 = interval(integer(5), integer(10))->set_complement(r3);
+    REQUIRE(r4->__str__() == "{x, y, z} \\ [5, 10]");
+
+    r3 = finiteset({symbol("x"), integer(7), symbol("z")});
+    r4 = interval(integer(5), integer(10))->set_complement(r3);
+    REQUIRE(r4->__str__() == "{x, z} \\ [5, 10]");
```

Comment:
Thanks. Can you open an issue on sympy repo on github?
Also, can you move the printing tests to `test_printing` and use symengine functions to test for equality instead of checking the string?